### PR TITLE
fix spotify_player build for macos

### PR DESCRIPTION
**Pull request by takac**
- add require Apple SDK libraries
- Fix dynamic linking to the sixel lib
- Only include alsa for linux builds, as this is not supported on MacOS

Tested on MacOS aarch64-darwin

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/audio/spotify-player/default.nix [hunk 1]
```diff
@@ -26,15 +29,25 @@ rustPlatform.buildRustPackage rec {
   nativeBuildInputs = [
     pkg-config
     cmake
+  ] ++ lib.optional stdenv.isDarwin [
+    rustPlatform.bindgenHook
```
Comment (dit7ya) [inline pkgs/applications/audio/spotify-player/default.nix:33]: I think it is alright to include `bindgenHook` for non-darwin too - for future-proofing.

**Review by dit7ya — APPROVED** [review]
LGTM.

File: pkgs/applications/audio/spotify-player/default.nix (hunk 1 above)
Comment (takac) [reply pkgs/applications/audio/spotify-player/default.nix:33]: Seems like it wasn't required for the Linux build, but will add it.

File: pkgs/applications/audio/spotify-player/default.nix [hunk 2]
```diff
@@ -8,6 +8,9 @@
 , dbus
 , fontconfig
 , libsixel
+, stdenv
+, darwin
+, makeWrapper
```
Comment (drupol) [inline pkgs/applications/audio/spotify-player/default.nix:13]: Can you please use `makeBinaryWrapper`, especially if you're looking to get it working on Darwin archs.
```suggestion
, makeBinaryWrapper
```

File: pkgs/applications/audio/spotify-player/default.nix [hunk 3]
```diff
@@ -26,15 +29,25 @@ rustPlatform.buildRustPackage rec {
   nativeBuildInputs = [
     pkg-config
     cmake
+    rustPlatform.bindgenHook
+  ] ++ lib.optional stdenv.isDarwin [
+    makeWrapper
```
Comment (drupol) [inline pkgs/applications/audio/spotify-player/default.nix:34]:
```suggestion
    makeBinaryWrapper
```

**Review by drupol — CHANGES_REQUESTED** [review]
Hi !

Thanks for your contribution.

I've left some comments, let me know if this is clear enough.

File: pkgs/applications/audio/spotify-player/default.nix [hunk 4]
```diff
@@ -44,6 +47,9 @@ rustPlatform.buildRustPackage rec {
   nativeBuildInputs = [
     pkg-config
     cmake
+    rustPlatform.bindgenHook
+  ] ++ lib.optional stdenv.isDarwin [
```
Comment (drupol) [inline pkgs/applications/audio/spotify-player/default.nix:51]:
```suggestion
  ] ++ lib.optionals stdenv.isDarwin [
```

**Review by drupol — CHANGES_REQUESTED** [review]

**Review by drupol — APPROVED** [review]

**Pull request merged**